enhancements/20220822-event-and-audit.md

```diff
@@ -0,0 +1,709 @@
+# Title
+
+Event and audit log 
+
+## Summary
+
+We need to provide a way for the user, especially the admin user to know what's happened in the system. E.g. who created which VM etc.
+
+As a part of this feature, we should implement an audit log mechanism as well.
+
+### Related Issues
+
+https://github.com/harvester/harvester/issues/578
+
+## Motivation
+
+### audit
+
+In `kubernetes`, the `audit` data is generated from `kube-apiserver`.
+
+https://kubernetes.io/docs/tasks/debug/debug-cluster/audit/
+
+```
+...
+Audit policy
+Audit policy defines rules about what events should be recorded and what data they should include. The audit policy object structure is defined in the audit.k8s.io API group. When an event is processed, it's compared against the list of rules in order. The first matching rule sets the audit level of the event. The defined audit levels are:
+
+None - don't log events that match this rule.
+Metadata - log request metadata (requesting user, timestamp, resource, verb, etc.) but not request or response body.
+Request - log event metadata and request body but not response body. This does not apply for non-resource requests.
+RequestResponse - log event metadata, request and response bodies. This does not apply for non-resource requests.
+```
+
+You can pass a file with the policy to `kube-apiserver` using the `--audit-policy-file` flag. If the flag is omitted, no events are logged. Note that the rules field must be provided in the audit policy file. A policy with no (0) rules is treated as illegal.
+
+### event
+
+Kubernetes `events` are objects that show you what is happening inside a cluster, such as what decisions were made by the scheduler or why some pods were evicted from the node. All core components and extensions (operators/controllers) may create events through the API Server.
+
+Events have no direct relationship with log messages generated by the various components, and are not affected with the log verbosity level. When a component creates an event, it often emits a corresponding log message. Events are garbage collected by the API Server after a short time (typically after an hour, 5 minutes in case of EKS), which means that they can be used to understand issues just happening, but you have to collect them to investigate past events.
+
+Events are the first thing to look at for application, as well as infrastructure operations when something is not working as expected. Keeping them for a longer period is essential if the failure is the result of earlier events, or when conducting post-mortem analysis.
+
+### Goals
+
+Collect `audit` log by default.
+Collect `event` log by default.
+Send `audit` and `event` log to `loki` by default, which can be queried by embedded `grafana`.
```

Comment:
The loki is not enabled by default, need to revise. Users can send logs to their data platform or provision a loki deployment.

Reply:
`loki` related parts are all moved to a separate chapter, thanks.